content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,313 @@
+
+To demonstrate sockets further, we shall use the Client/Server architecture. Client and server communicate by writing to and reading from the socket connection. The machines must have each other's network information to make this connection.
+
+### Table of contents:
+- [Definition of a Socket](#definition-of-a-socket)
+- [Programming on the Server-side ](#programming-on-the-server-side)
+- [Programming on the Client-side](#programming-on-the-client-side)
+- [How to Run Programs in a Variety of Applications](#how-to-run-programs-in-a-variety-of-applications)
+- [Conclusion](#conclusion)
+
+### Definition of a Socket
+
+A Socket is a communication **endpoint** that serves as a link between two machines on a network. It has a socket number, which the TCP/IP layer can use to identify the application that receives the data. Additionally, a port number and an IP address are what make up an endpoint.
+
+#### What is TCP?
+Transmission Control Protocol (TCP) - This is one of the most important protocols of the Internet Protocols suite. It is a widely used protocol for data transmission in a network that operates in Client/Server point-to-point mode.
+
+#### Two categories of Sockets:
+
+1. A server socket awaits a request from a client.
+2. Client socket - utilized for sending and receiving requests from clients.
+
+#### The client has to know two things about the server:
+
+1. Its server's IP address 
+2. Its port number
+
+>The ports 0-1024 are for administrative use only (e.g.  21 for FTP, 23 for Telnet, 25 for email, and 80 for HTTP). In our program, we'll be using port number 5000.
+
+#### When we make a socket connection in the Java programming language, we go through the following steps:
+
+- The Server constructs a `ServerSocket` object to specify the port number on which our conversation will occur. Exception handling methods are in use whenever an I/O error occurs. **Throws** an **I/OException** or a **try** **catch** block is used.
+
+- The `accept()` function is called by the server. This method waits for a client to connect to the server on a specific port before returning.
+
+- A client creates a `Socket` object, giving the server name and the connecting port number as the server awaits.
+
+- The `Socket` class constructor will attempt to connect the client to the server and port number provided. The client now has a socket object that can connect to the server if communication occurs.
+
+- The server's `accept()` method returns a reference to a new server socket connected to the client's socket.
+ 
+>If the connection is successful, the client and server can then communicate using **I/O streams**. The client and server socket classes provide the **I/O streams**. The client's OutputStream communicates with the server's InputStream, and the server's OutputStream communicates with the client's InputStream.
```

Comment:
Make this part not a blockquote, it does not seem necessary (i.e., remove the \> before *"If the connection is successful..."*